File: program/program/find-average-of-numbers-by-recursion/find_average_of_numbers_by_recursion.dart
import 'dart:io';

double findAverage(List<int> numbers, int index, double sum) {
  // Base case: when index reaches the end of the list
  if (index == numbers.length) {
    // Return the average
    return sum / numbers.length;
  } else {
    // Recursive case: add current number to sum and move to the next index
    return findAverage(numbers, index + 1, sum + numbers[index]);
  }
}

void main() {
  print("Enter the numbers separated by space:");
  String input = stdin.readLineSync()!;
  List<String> numberStrings = input.split(' ');
  
  // Convert input strings to integers
  List<int> numbers = numberStrings.map(int.parse).toList();
  
  // Calculate the average using recursion
  double average = findAverage(numbers, 0, 0);
  
  print("Average of the numbers: $average");
}

File: program/program/find-average-of-numbers-by-recursion/findAverageOfNumbersByRecursion.scala
//Write a Scala program to find average of numbers by recursion 

object AverageCalculator {
  def main(args: Array[String]): Unit = {
    println("Enter numbers separated by space:")
    val input = scala.io.StdIn.readLine()
    val numbers = input.split(" ").map(_.toInt).toList

    if (numbers.nonEmpty) {
      val average = calculateAverage(numbers)
      println(s"Average: $average")
    } else {
      println("Please enter at least one number.")
    }
  }

  def calculateAverage(numbers: List[Int]): Double = {
    def helper(list: List[Int], sum: Int, count: Int): Double = {
      list match {
        case Nil => sum.toDouble / count
        case head :: tail => helper(tail, sum + head, count + 1)
      }
    }
    helper(numbers, 0, 0)
  }
}